### PR TITLE
blog: april commit of the month

### PR DESCRIPTION
**Pull request by jazzdan**

### CODE REVIEW

File: blog/_posts/2020-05-04-april-commit-of-the-month.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  - tilt
+  - pack
+  - buildpack
+---
+
+In April we introduced [Tilt Extensions](https://docs.tilt.dev/extensions.html). Extensions are open-source packaged functions that extend the capability Tilt, right inside your Tiltfile. Since releasing them we've seen several great extensions contributed by members of the community and we're highlighting one of them as April's Commit of the Month: [an extension to use pack and buildbacks to build container images](https://github.com/windmilleng/tilt-extensions/commit/e1d193e508ce8468800d0985ee4714aa65d78c87) by [Gareth Rushgrove](https://twitter.com/garethr)!
+
+## What is buildpack? What is pack?
+
+From [the docs](https://buildpacks.io/docs/concepts/components/buildpack/): A buildpack is a unit of work that inspects your app source code and formulates a plan to build and run your application.
```
Comment (maiamcc) [inline blog/_posts/2020-05-04-april-commit-of-the-month.md:25]: huh, neat! Could you give an example here of what a trivial buildpack might look like? Would help with comprehension imho

File: blog/_posts/2020-05-04-april-commit-of-the-month.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+- `buildpack.toml` – provides metadata about your buildpack
+- `bin/detect` – determines whether buildpack should be applied
+- `bin/build` – executes buildpack logic
+
+`pack` is a command line tool that takes your source code and invokes the appropriate buildpack(s) to create a Docker image, or deploy it straight in to the cloud of your choice.
+
+## How does the Tilt pack extension work?
+
+This extension adds the `pack` function, which is used the same way as `docker_build` to allow Tilt
```
Comment (maiamcc) [inline blog/_posts/2020-05-04-april-commit-of-the-month.md:37]:
```suggestion
This extension makes the `pack` function available in your Tiltfile, which is used the same way as `docker_build` to allow Tilt
```
maybe this is gratuitous, but seems useful explanation to someone who doesn't yet grok extensions

File: blog/_posts/2020-05-04-april-commit-of-the-month.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  - extensions
+keywords:
+  - tilt
+  - pack
+  - buildpack
+---
+
+In April we introduced [Tilt Extensions](https://docs.tilt.dev/extensions.html). Extensions are open-source packaged functions that extend the capability Tilt, right inside your Tiltfile. Since releasing them we've seen several great extensions contributed by members of the community and we're highlighting one of them as April's Commit of the Month: [an extension to use pack and buildbacks to build container images](https://github.com/windmilleng/tilt-extensions/commit/e1d193e508ce8468800d0985ee4714aa65d78c87) by [Gareth Rushgrove](https://twitter.com/garethr)!
+
+## What is buildpack? What is pack?
```
Comment (maiamcc) [inline blog/_posts/2020-05-04-april-commit-of-the-month.md:23]: should this be:
- what is `buildpack`
- what is a buildpack
- what is a `buildpack`
- what are buildpacks?

I feel like `pack` as a CLI tool should always be in backticks, here and in the header below

File: blog/_posts/2020-05-04-april-commit-of-the-month.md (hunk 1 above)
Comment (jazzdan) [reply blog/_posts/2020-05-04-april-commit-of-the-month.md:25]: I might link to it but TBH it's a bit confusing so I'd rather not dive in to it here =\

File: blog/_posts/2020-05-04-april-commit-of-the-month.md (hunk 3 above)
Comment (jazzdan) [reply blog/_posts/2020-05-04-april-commit-of-the-month.md:23]: ah yeah good call

File: blog/_posts/2020-05-04-april-commit-of-the-month.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+`pack` is a command line tool that takes your source code and invokes the appropriate buildpack(s) to create a Docker image, or deploy it straight in to the cloud of your choice.
+
+## How does the Tilt pack extension work?
+
+This extension makes the `pack` function available in your Tiltfile, which is used the same way as `docker_build` to allow Tilt
+to automatically build a container image with a known name.
+
+```python
+pack('example-image')
```
Comment (nicks) [inline blog/_posts/2020-05-04-april-commit-of-the-month.md:41]: for what it's worth - one of the big lessons I had from working on textbooks and o'reilly books was that all code snippets should have link to a working example. You'd be amazed at how often people write totally broken examples without something to check them against, even for 1-2 line code snippets

Comment (nicks) [reply blog/_posts/2020-05-04-april-commit-of-the-month.md:41]: in this case, I might just link to the README and use his exact snippet

```
load('ext://pack', 'pack')

pack('example-image')
k8s_yaml('kubernetes.yaml')
k8s_resource('example-deployment', port_forwards=8000)
```

File: blog/_posts/2020-05-04-april-commit-of-the-month.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  - tilt
+  - cotm
+  - extensions
+keywords:
+  - tilt
+  - pack
+  - buildpack
+---
+
+In April we introduced [Tilt Extensions](https://docs.tilt.dev/extensions.html). Extensions are open-source packaged functions that extend the capability Tilt, right inside your Tiltfile. Since releasing them we've seen several great extensions contributed by members of the community and we're highlighting one of them as April's Commit of the Month: [an extension to use pack and buildbacks to build container images](https://github.com/windmilleng/tilt-extensions/commit/e1d193e508ce8468800d0985ee4714aa65d78c87) by [Gareth Rushgrove](https://twitter.com/garethr)!
```
Comment (ellenkorbes) [inline blog/_posts/2020-05-04-april-commit-of-the-month.md:21]: s/capability Tilt/capability of Tilt

**Pull request merged**